<launch>

  <!-- Produce task space motion (figure of eight) -->
  <node pkg="ik_ros_examples" name="figure_eight_motion" type="figure_eight_node.py">
    <param name="parent_frame_id" type="string" value="figure_eight_base"/>
    <param name="child_frame_id" type="string" value="figure_eight"/>
  </node>

  <node pkg="tf2_ros" type="static_transform_publisher" name="figure_eight_base_in_world"
	args="0.5 0 0.5 0 1.57 0 world figure_eight_base"/>

  <!-- Parse figure of eight to IK setup -->
  <node pkg="ik_ros" name="tf_to_floatarray_figure_eight" type="tf_to_floatarray_node.py">
    <param name="parent_frame_id" type="string" value="world"/>
    <param name="child_frame_id" type="string" value="figure_eight"/>
    <remap from="transform" to="setup"/>
  </node>

  <!-- IK Node -->
  <node pkg="ik_ros" name="ik_node" type="ik_node.py" args="SingleArmExoticaEffFrame" output="screen">
    <param name="exotica_xml_filename" type="string" value="{ik_ros_examples}/configs/lwr.xml"/>
    <param name="target_frame_id" type="string" value="Target"/>
    <param name="base_frame_id" type="string" value="base"/>
    <remap from="joint_states/target" to="rpbi/lwr/joint_state/target"/>
  </node>

  <!-- ROS-Pybullet interface node simulator -->
  <node pkg="ros_pybullet_interface" name="ros_pybullet_interface_node" type="ros_pybullet_interface_node.py">

    <!-- Config file for Pybullet -->
    <param name="pybullet_config" value="{ros_pybullet_interface_examples}/configs/pybullet_basic_example.yaml" type="str"/>

    <!-- Node frequency -->
    <param name="ros_node_freq" value="100" type="int"/>

    <!-- Simulated robot -->
    <rosparam param="robot_config_file_names">
      [
      "{ik_ros_examples}/configs/lwr.yaml",
      ]
    </rosparam>

    <rosparam param="visual_object_config_file_names">
      [
      "{ik_ros_examples}/configs/sphere.yaml"
      ]
    </rosparam>

  </node>

  <node pkg="tf2_ros" type="static_transform_publisher" name="simulated_robot_base_tf_broadcaster"
	args="0.0 0.0 0.0 0 0 0 rpbi/world lwr_base"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="attach_world_to_rpbi_world"
	args="0 0 0 0 0 0 world rpbi/world"/>

</launch>
